spec/lib/traffic_source_parser/parser/referrer_parser_spec.rb

```diff
@@ -0,0 +1,217 @@
+require 'spec_helper'
+
+describe TrafficSourceParser::Parser::ReferrerParser do
+
+  shared_examples_for "returns the correct object with attributes" do
+    it "returns klass object with hash attributes" do
+       sources.each do |source|
+          cookie_value = source.delete(:cookie_value)
+          parsed_result = TrafficSourceParser.parse(cookie_value)
+          expect(parsed_result).to be_a klass
+          source.keys.each do |key|
+            expect(parsed_result.send(key)).to eq source[key]
+          end
+        end
+    end
+  end
+
+  describe ".parse" do
+
+    context "when cookie value is (none)" do
+
+      let(:sources) do
+        [
+          {
+            cookie_value: "(none)",
+            source: "(direct)",
+            campaign: "(direct)",
+            channel: "Direct",
+            medium: "(none)"
+          }
+        ]
+      end
+
+      let(:klass) {  TrafficSourceParser::Result::Direct }
+
+      it_behaves_like "returns the correct object with attributes"
+
+    end
+
+    context "when cookie value is empty" do
+
+      let(:sources) do
+        [
+          {
+            cookie_value: '',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.